packages/language-server/src/plugins/typescript/features/FindComponentUsagesProvider.ts

```diff
@@ -0,0 +1,127 @@
+import { Location, Position, Range } from 'vscode-languageserver';
+import { URI } from 'vscode-uri';
+import { flatten, pathToUrl } from '../../../utils';
+import { FindComponentUsagesProvider } from '../../interfaces';
+import { LSAndTSDocResolver } from '../LSAndTSDocResolver';
+import { convertToLocationRange, hasNonZeroRange } from '../utils';
+import { Document, DocumentManager } from '../../../lib/documents';
+import ts from 'typescript';
+import { isNoTextSpanInGeneratedCode, SnapshotFragmentMap } from './utils';
+import { SvelteSnapshotFragment } from '../DocumentSnapshot';
+import { lsConfig } from '../../../ls-config';
+export class FindComponentUsagesProviderImpl implements FindComponentUsagesProvider {
+    constructor(private readonly lsAndTsDocResolver: LSAndTSDocResolver) {}
+
+    async findComponentUsages(uri: string): Promise<Location[] | null> {
+        const u = URI.parse(uri);
+        const fileName = u.fsPath;
+        const document = await this.getDocument(fileName);
+        const { lang, tsDoc } = await this.getLSAndTSDoc(document);
+        const fragment = tsDoc.getFragment();
+        const ignoreImports =
+            lsConfig.getConfig().typescript.findComponentUsagesIgnoresImports.enable;
+
+        const position = this.GetClassPosition(fragment);
+
+        const references = lang.findReferences(tsDoc.filePath, fragment.offsetAt(position));
+
+        if (!references) {
+            return null;
+        }
+
+        const docs = new SnapshotFragmentMap(this.lsAndTsDocResolver);
+        docs.set(tsDoc.filePath, { fragment, snapshot: tsDoc });
+
+        const locations = await Promise.all(
+            flatten(references.map((ref) => ref.references))
+                .filter((ref) => !ref.isDefinition)
+                .filter(this.notInGeneratedCode(tsDoc.getFullText()))
+                .map(async (ref) => {
+                    const defDoc = await docs.retrieveFragment(ref.fileName);
+                    const document = await this.getDocument(ref.fileName);
```

Comment:
You can't assume it's a Svelte document here (DocumentManager only handles these, not very clear from the name I realize now), you need to use fragment/snapshot similar to other code here - but it will be irrelevant anyway if you delete the "filter out imports" code I assume

Reply:
I would still need this to filter out the end tag(which should only be svelte docs) but this had been working for .TS files just fine. Did you expect it not to work? Also, I dont understand the difference between the different flavors of "document" 

What would be a good explanation of...
- Document
- DocumentSnapshot
- SvelteDocumentSnaphot
- SvelteSnapshotFragment
- TextDocumentItem
- JSOrTSDocumentSnapshot
- SnapshotFragment